pkg/ingress/tls.go

```diff
@@ -89,6 +102,7 @@ func (i *Tls) newCertNeeded() bool {
 	minimumValidity := i.ingress.KubeLego().LegoMinimumValidity()
 	timeLeft := expireTime.Sub(time.Now())
 	logger := i.Log().WithField("expire_time", expireTime)
+	certRenewalTime.WithLabelValues(strings.Join(i.Hosts(), ",")).Set(float64(timeLeft))
```

Comment:
I think this is a bad idea, we should just report the expire time instead of the time left. The expire time is stable and not depending on `time.Now()`

Reply:
I guess the expire time is stable, but how would we export that a prometheus counter? a counter works as you can easily create graphs and alarms for say, under 20 days?

Reply:
You can do the calculation in prometheus:

> `ExpireTime - Now  <= 10d` (pseudo-code)

The node exporter exports the boot_time like that:

https://github.com/prometheus/node_exporter/blob/859a825bb84bf4fb911fcba445d410965945f088/collector/stat_linux.go#L64

We need a gauge here because in theory the boot time can go in any direction

Reply:
add, didn't know that! I will fix